pkg/kubelet/prober/prober.go

```diff
@@ -200,13 +202,19 @@ func (pb *prober) runProbe(p *api.Probe, pod *api.Pod, status api.PodStatus, con
 		return pb.exec.Probe(pb.newExecInContainer(pod, container, containerID, p.Exec.Command))
 	}
 	if p.HTTPGet != nil {
+		scheme := p.HTTPGet.Scheme
+		host := p.HTTPGet.Host
+		if host == "" {
+			host = status.PodIP
+		}
 		port, err := extractPort(p.HTTPGet.Port, container)
 		if err != nil {
 			return probe.Unknown, "", err
 		}
-		host, port, path := extractGetParams(p.HTTPGet, status, port)
-		glog.V(4).Infof("HTTP-Probe Host: %v, Port: %v, Path: %v", host, port, path)
-		return pb.http.Probe(host, port, path, timeout)
+		path := p.HTTPGet.Path
+		glog.V(4).Infof("HTTP-Probe Host: %v%v, Port: %v, Path: %v", scheme, host, port, path)
+		url := formatURL(string(scheme), host, port, path)
```

Comment:
I meant to ToLower() the scheme.  You're taking the api constant, "HTTP" and casting it to a string, resulting in HTTP://whatever:1234.

I just realized I did not send that comment elsewhere.  API constants should be capitalized, thus HTTP and HTTPS.  It's weird but consistent with the whole API.